.github/workflows/Docker.yaml

```
@@ -0,0 +1,80 @@
+name: Docker
+on:
+  pull_request:
+    paths:
+      - "**"
+      - "!docs/**"
+      - "!README.md"
+      - "!CONTRIBUTING.md"
+      - ".github/workflows/Docker.yaml"
+      - "!mkdocs.yml"
+      - "!pyproject.toml"
+      - "!poetry.lock"
+  workflow_dispatch:
+    inputs:
+      version:
+        description: version of the scenario_simulator_v2
+        required: true
+
+jobs:
+  push_docker:
+    name: Push Docker Image
+    runs-on: ubuntu-22.04
+    timeout-minutes: 720
+    strategy:
+      matrix:
+        rosdistro: [humble]
+        arch: [amd64]
+        # Build test for arm64 CPU is broken.
+        # This is a temporary solution and will be repaired in the future.
+        # See also https://github.com/tier4/scenario_simulator_v2/pull/1295
+        # arch: [amd64, arm64]
+    steps:
+      - name: Free Disk Space (Ubuntu)
+        uses: jlumbroso/free-disk-space@main
+        with:
+          tool-cache: false
+
+      - name: Install docker for ubuntu runner
+        uses: docker/setup-buildx-action@v3
+
+      - name: Install QEMU
+        uses: docker/setup-qemu-action@v3
+
+      - uses: actions/checkout@v3
```

This is a good opportunity to use the latest action.
```suggestion
      - uses: actions/checkout@v4
```